app/src/main/AndroidManifest.xml

```diff
@@ -1,24 +1,30 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
   xmlns:tools="http://schemas.android.com/tools">
+
   <uses-permission android:name="android.permission.INTERNET" />
-  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"
+    android:maxSdkVersion="32" />
   <uses-permission android:name="android.permission.READ_SYNC_SETTINGS" />
   <uses-permission android:name="android.permission.READ_SYNC_STATS" />
   <uses-permission android:name="android.permission.REORDER_TASKS" />
   <uses-permission android:name="android.permission.WRITE_SYNC_SETTINGS" />
-  <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+  <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"
+    android:maxSdkVersion="29"/>
   <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
   <uses-permission android:name="android.permission.AUTHENTICATE_ACCOUNTS" />
   <uses-permission android:name="android.permission.GET_ACCOUNTS" />
   <uses-permission android:name="android.permission.USE_CREDENTIALS" />
   <uses-permission android:name="android.permission.MANAGE_ACCOUNTS" />
   <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
-  <uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>
+  <uses-permission android:name="android.permission.READ_MEDIA_IMAGES"
+    android:minSdkVersion="33"/>
   <uses-permission android:name="com.google.android.apps.photos.permission.GOOGLE_PHOTOS" />
   <uses-permission android:name="android.permission.SET_WALLPAPER"/>
   <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
```

Comment:
We use foreground service in our app to ensure contributions get uploaded reliably even when users are using 'Optimised' battery mode. Targeting Android 14 makes it [mandatory to specify a foreground service type](https://developer.android.com/about/versions/14/behavior-changes-14#fgs-types). Could you check this once?

Reply:
Yes, I checked that but I couldn't find that service in the `Manifest` file, and also don't know what type of service should I assign to it. Only two services are defined in the Manifest file. Can you help me with that?

Reply:
We are using WorkManager to show the notification while uploading contributions. So, we don't need to specify the type of foreground service when using WorkManager.
However, two services are defined in the AndroidManifest file and SDK Upgrade Assistant suggests specifying a type for them. Those services are not foreground services, I guess. What do you think?

Reply:
I haven't gone through the complete documentation yet, but would you mind checking https://developer.android.com/develop/background-work/services/fg-service-types#data-sync and sharing your thoughts?

Reply:
Thank you @RitikaPahwa4444 for referencing these docs. 
As you said earlier, we are using foreground service for showing the uploading status but I got confused because we didn't have any foreground service defined in the manifest file.
WorkManager automatically utilizes a system-provided foreground service for showing notifications. I am adding the type for this service. Do we have more foreground services except this one?

Reply:
> we didn't have any foreground service defined in the manifest file.

Could you please elaborate? As far as I remember, this existed way before we added it to the upload worker.

Reply:
Which one is a foreground service? Because these are the only two services defined in the `Manifest` file.

     <service
      android:name=".auth.WikiAccountAuthenticatorService"
      android:exported="true"
      android:process=":auth">
      <intent-filter>
        <action android:name="android.accounts.AccountAuthenticator" />
      </intent-filter>
      <meta-data
        android:name="android.accounts.AccountAuthenticator"
        android:resource="@xml/authenticator" />
    </service>

    <service
      android:name="org.acra.sender.SenderService"
      android:exported="false"
      android:process=":acra" />